WordPress/playground-tools
# Playground block UX improvements

## What?

* Updates Playground packages versions
* Wraps modal forms with HTML `<form>` tags to make the enter key work and adds autofocus
* Wraps `activatePlugin` and `handleCodeInjection` with try/catch to prevent crashes in the editor (TODO – investigate the root cause of these crashes)

Demo:


https://github.com/WordPress/playground-tools/assets/205419/5f0b794b-866a-4587-acfe-939e690de625



## Remaining work

- [ ] In Playground – limit the dark background to the loading screen – let's tackle that separately
- [x] In Playground – fix the `getCurrentURL` method, accept current URL in the `goTo` method
- [x] In Playground – https://github.com/WordPress/wordpress-playground/pull/902

## Testing Instructions

Run the development build of the Playground block:

```
nx dev wordpress-playground-block
```

This will start:

-   A webpack development server for the `wordpress-playground-block` package
-   A new WordPress site with the `wordpress-playground-block` plugin activated

From here, play with the block, confirm the code editor works, the data is saved and rendered as expected, and that the block works in general.

[wordpress-playground-block.zip](https://github.com/WordPress/playground-tools/files/13873566/wordpress-playground-block.zip)


cc @dawidurbanski
